### ISSUE TITLE
support tuplets other than 3let and 6let

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#74]
Actually Audiveris only supports triplets and sixlets(?) as tuplets.
There are a couple of more, and there is one that appears rather often, too: duplets.
It means that there are 2 notes in a place of normally 3, esp. in a 3/4 (6/8, 3/8) time signature.

It's a rather seldom case and should therefore be kept in mind for the future...

**Comment by maximumspatium** [comment]
There are not only triplets/sixlets in music. But also rare 2-lets, 5-lets, 7-lets, 9-lets, 10-lets, 11-lets, 12-lets and even more. They are usually single and signed.
![image](https://user-images.githubusercontent.com/36125949/57545149-236f5980-7362-11e9-801a-6eecdd1465c1.png)
![image](https://user-images.githubusercontent.com/36125949/57544146-29b00680-735f-11e9-9bcd-3b977cb4bf80.png)
![image](https://user-images.githubusercontent.com/36125949/57544308-a3e08b00-735f-11e9-8a30-ae3cb3b84a16.png)
![image](https://user-images.githubusercontent.com/36125949/57544544-557fbc00-7360-11e9-8f06-dcb75f8a7347.png)
![dic888](https://user-images.githubusercontent.com/36125949/57544697-d2ab3100-7360-11e9-9176-beeaa3366183.jpeg)
They are popular in Chopin music. It has even a 22-let.
![image](https://user-images.githubusercontent.com/36125949/57544787-16059f80-7361-11e9-87a3-0fe4535b13cc.png)

**Comment by maximumspatium** [comment]
@oleedd @Bacchushlg  @hbitteur 

The easiest way to handle N-tuplets is via a dedicated UI control as follows:

1. click the N-tuplet symbol in the palette on the right (we need to add such a symbol first)
2. drag a lasso to select a group of notes to be converted to tuplet like that:
![N-tuplet selection](https://user-images.githubusercontent.com/6561800/57567928-b5b64280-73e0-11e9-8d48-da8b66ea16ba.png)
3. a dialog appears allowing to specify the nominator (number of notes in the tuplet) and the denominator (tuplet time) values. In the above example this would be 7:8.

4. (optional) we can offer a simple search functionality for similar groups, à la "search & replace" found in text processors. The user would need to verify and confirm the results and Audiveris will fix them all in a batch.

Although the above proposal distracts from the existing UI for handling triplets and sixlets, it's much more convenient and flexible than the way we currently handle that: the "3"/"6" must be dragged to the right position starting in the middle of the staff in the middle of the group etc.

We can discuss the tuplet UI unification as well...

**Comment by oleedd** [comment]
I agree. But that "N-tuplet symbol" button should remember the last chosen value.
What does "8" mean in "7;8" if in normal rhythm there should be 4 thirty-second notes?

**Comment by oleedd** [comment]
I agree. But that "N-tuplet symbol" button should remember the last chosen value.

**Comment by oleedd** [comment]
I agree. But that "N-tuplet symbol" button should remember the last chosen value. So maybe better to chose a value before dragging. Something like `<input type="number">` in html is need.
